src/actions.rs

```diff
@@ -188,6 +188,12 @@ impl ActionHandler {
         self.build_current_project(out);
     }
 
+    pub fn on_save(&self, save: DidSaveTextDocumentParams, out: &Output) {
+        let fname: PathBuf = parse_file_path(&save.text_document.uri).unwrap();
+        self.vfs.file_saved(&fname).unwrap();
+        self.build_current_project(out);
```

Comment:
I think that this should have BuildPriority::Immediate, rather than Normal (not 100% sure about this, but I think it would be expected), easiest way to do this is for `build_current_project` to take a priority argument.